### DISCUSSION TITLE
grammer and parser of query expression

### DISCUSSION THREAD

**Discussion by gaoyw1** [milvus-io/milvus#34227] [Q&A and General discussion]
Hi, where can i find the document about expression grammer of query operation?  milvus source code of expression parsing shows that it is generated, i want to know how to generate it? which tools is used to generate this code? thanks a lot

**Comment by yhmo** [comment]
Scalar Filtering Rules:
https://milvus.io/docs/boolean.md

Filtered search:
https://milvus.io/docs/single-vector-search.md#Filtered-search

Filtered query:
https://milvus.io/docs/get-and-scalar-query.md

Filtering on Array field:
https://milvus.io/docs/array_data_type.md#Basic-scalar-filtering

Filtering on JSON field:
https://milvus.io/docs/use-json-fields.md#Basic-scalar-filtering


Source code to parse expression:
https://github.com/milvus-io/milvus/tree/master/internal/core/src/query
